infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -463,7 +465,7 @@ private void fetchTopologyFromClusterManager() {
      * Stop Log Replication
      */
     private void stopLogReplication() {
-        if (localClusterDescriptor.getRole() == ClusterRole.ACTIVE && isLeader.get()) {
+        if (localClusterDescriptor != null && localClusterDescriptor.getRole() == ClusterRole.ACTIVE && isLeader.get()) {
```

Comment:
A little confused, If a new topology is not valid, we don't update the localClusterDescriptor, right? How we hit this NPE?

If the localClusterDescriptor is always null, how does the log replication start?... It should hit NPE here.
https://github.com/CorfuDB/CorfuDB/blob/8fc1b8b4671c8783462330247c65ceebfa9bef07/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java#L414

Reply:
Yes,  we don't update the localClusterDescriptor and this is called from:

https://github.com/CorfuDB/CorfuDB/blob/8fc1b8b4671c8783462330247c65ceebfa9bef07/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java#L579

At this point we haven't started LR, we wouldn't even be able to acquire the lock because we are still not recognized as part of any cluster...We just. get topology updates from SM, where we don't show up as part of any active or standby cluster...